### PR TITLE
Including eventing-kafka upgrade tests for source and channel

### PR DESCRIPTION
**Pull request by cardil**
Port of #983 to main branch with 0.22 eventing

This PR requires also https://github.com/openshift/release/pull/19133 to be merged

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -41,7 +41,7 @@ replace (
 	k8s.io/client-go => k8s.io/client-go v0.19.7
 	k8s.io/code-generator => k8s.io/code-generator v0.19.7
 	k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20200805222855-6aeccd4b50c6
-	knative.dev/eventing => github.com/openshift/knative-eventing v0.99.1-0.20210518171156-3b8745c96673
-	knative.dev/eventing-kafka => github.com/openshift-knative/eventing-kafka v0.19.1-0.20210504074514-a18051f72852
+	knative.dev/eventing => github.com/cardil/eventing v0.11.1-0.20210609102549-481a26fd9454
+	knative.dev/eventing-kafka => github.com/cardil/eventing-kafka v0.21.1-0.20210609130636-52e5100c8a0b
```
Comment (cardil) [inline go.mod:45]: /hold

This PR requires https://github.com/openshift/knative-eventing/pull/1309 and https://github.com/openshift-knative/eventing-kafka/pull/233 to be merged first, and the interim personal forks removed.

File: test/lib.bash [hunk 2]
```diff
@@ -193,17 +193,16 @@ function run_rolling_upgrade_tests {
   patch="${KNATIVE_SERVING_HOME}/openshift/patches/001-object.patch"
   git apply "${patch}"
 
-  image_version=$(versions.major_minor "${KNATIVE_SERVING_VERSION}")
+  # FIXME(ksuszyns): remove static version resolving
+  image_version=0.21
```
Comment (cardil) [inline test/lib.bash:197]: This is interim change, to be removed when https://github.com/openshift/release/pull/19133 is merged and images are mirrored

**Comment by cardil** [comment]
Here's a passing build on latest commit: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/1011/pull-ci-openshift-knative-serverless-operator-main-4.7-upgrade-tests-aws-ocp-47/1402743138010796032

Other failures are infra and eventing-kafka source test flakiness.

File: go.mod (hunk 1 above)
Comment (cardil) [reply go.mod]: Personal forks has been removed

File: test/lib.bash (hunk 2 above)
Comment (cardil) [reply test/lib.bash]: openshift/release#19133 is merged and images are mirrored

/unhold

**Comment by cardil** [comment]
/retest

Looks like eventing flakes, not associated with this PR.

**Comment by cardil** [comment]
Merging #1010 with 21088d4 to help with eventing kafka CR flakes

**Comment by mgencur** [comment]
/lgtm
/hold
Putting on hold until all the checks pass. Feel free to unhold when they pass.

**Pull request merged**